File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans; // We intialized a vector of vector ans to return
        if(root == NULL){ //If there's nothing return empty vector
            return ans;
        }

        queue<TreeNode*> q; // Initialized a Queue
        q.push(root);

        while(!q.empty()){
            int size = q.size();
            vector<int> level;
            for(int i=0;i<size;i++){
                TreeNode *node = q.front();
                q.pop();
                if(node->left!=NULL) {
                    q.push(node->left);
                }
                if(node->right!=NULL) {
                    q.push(node->right);
                }
                level.push_back(node->val);

            }
            ans.push_back(level);

        }
        return ans;
    }
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        q = deque([root])
        res = []
        while q:
            res.append([i.val for i in q])
            size = len(q)
            for i in range(size):
                cur = q.popleft()
                if (cur.left):
                    q.append(cur.left)
                if (cur.right):
                    q.append(cur.right)


        return res